src/lsm/grid.zig

```diff
@@ -229,34 +229,32 @@ pub fn GridType(comptime Storage: type) type {
             for (grid.read_iop_tracer_slots) |slot| assert(slot == null);
             for (grid.write_iop_tracer_slots) |slot| assert(slot == null);
 
-            grid.cache_valid.deinit(allocator);
-            grid.cache.deinit(allocator);
-
             for (grid.cache_blocks) |block| allocator.free(block);
             allocator.free(grid.cache_blocks);
 
+            grid.cache_valid.deinit(allocator);
+            grid.cache.deinit(allocator);
+
             grid.* = undefined;
         }
 
         pub fn cancel(grid: *Grid, callback: fn (*Grid) void) void {
-            assert(!grid.cancelling);
-            assert(grid.cancel_callback == null);
+            assert(grid.canceling == null);
 
-            grid.cancelling = true;
-            grid.cancel_callback = callback;
+            grid.canceling = .{ .callback = callback };
```

Comment:
This is a nice fallout. Didn't realize we would get this as well out of the change.